src/validators/file_presence.rs

```diff
@@ -0,0 +1,71 @@
+use crate::validators::issues::{Issue, IssueType, Severity};
+
+const MANDATORY_FILES: &[&str] = &[
+    "agency.txt",
+    "calendar.txt",
+    "calendar_dates.txt",
+    "routes.txt",
+    "stops.txt",
+    "stop_times.txt",
+    "trips.txt",
+];
+
+const OPTIONAL_FILES: &[&str] = &[
+    "fare_attributes.txt",
+    "fare_rules.txt",
+    "feed_info.txt",
+    "frequencies.txt",
+    "shapes.txt",
+];
+
+fn missing_files(raw_gtfs: &gtfs_structures::RawGtfs) -> Vec<Issue> {
+    MANDATORY_FILES
+        .iter()
+        .filter(|m| !raw_gtfs.files.iter().any(|f| f.ends_with(*m)))
+        .map(|m| {
+            Issue::new(Severity::Fatal, IssueType::MissingMandatoryFile, m)
+                .details("The mandatory file was not found")
+        })
+        .collect()
+}
+
+fn extra_files(raw_gtfs: &gtfs_structures::RawGtfs) -> Vec<Issue> {
+    raw_gtfs
+        .files
+        .iter()
+        .filter(|f| {
+            !MANDATORY_FILES.iter().any(|m| f.ends_with(m))
+                && !OPTIONAL_FILES.iter().any(|o| f.ends_with(o))
+        })
+        .map(|f| {
+            Issue::new(Severity::Error, IssueType::ExtraFile, f)
+                .details("The mandatory file was not found")
+        })
+        .collect()
+}
+
+pub fn validate(raw_gtfs: &gtfs_structures::RawGtfs) -> Vec<Issue> {
+    missing_files(raw_gtfs)
+        .into_iter()
+        .chain(extra_files(raw_gtfs).into_iter())
+        .collect()
+}
+
+#[test]
+fn test_missing() {
+    let raw = gtfs_structures::RawGtfs::new("test_data/missing_mandatory_files").unwrap();
+    let validations = missing_files(&raw);
+    assert_eq!(1, validations.len());
+    assert_eq!(IssueType::MissingMandatoryFile, validations[0].issue_type);
+    assert_eq!("stop_times.txt", validations[0].object_id);
+    assert_eq!(Severity::Fatal, validations[0].severity);
+}
+
+#[test]
```

Comment:
can you add a test on the extra files ?

Reply:
what’s wrong `with test_extra` ?

Reply:
oups my bad, I wrongly read the test :roll_eyes: